docs/pages/docs/core-concepts/caching.mdx

```diff
@@ -7,9 +7,44 @@ import Callout from "../../../components/Callout";
 
 # Caching
 
-Unlike other tools you may have used, Turborepo can cache emitted files and logs of previously run commands, so it can skip work that has already been done, yielding incredible time savings.
+Turborepo can cache the results and logs of any task in your codebase—meaning you never need to do the same work twice.
 
-By default, [`turbo run`](../reference/command-line-reference#turbo-run-task1-task2-1) considers any files emitted in the `dist/**` or `build/**` folders of a workspace task (defined in that workspace's `package.json` `scripts` object) to be cacheable. In addition, `turbo run` treats logs (`stderr` and`stdout`), which are automatically written to `.turbo/run-<command>.log`, as a cacheable artifact. By treating logs as artifacts, you can cache just about any command in your Turborepo.
+## Missing the cache
+
+Each task in your codebase has **inputs** and **outputs**.
+
+- A `build` task might take in **source files as input** and return **bundled files as output**.
+- A `lint` or `test` task might take in **source files**, and only output to **`stdout`**.
+
+Let's say you run a `build` task with Turborepo using `turbo run build`:
+
+![](../../../public/images/docs/cache-miss.png)
+
+1. Turborepo will **take the inputs to your task** (by default all non-gitignored files in the workspace folder), and **turn them into a hash**, like `78awdk123`.
+
+2. It'll then **check its cache** for a folder named `78awdk123`.
+
+3. If it doesn't find it, it **runs the task**.
+
+4. Once the task is over, it **saves all the outputs** (including logs) into its cache under the hash.
```

Comment:
4. Once the task is over, it **saves all the outputs** (including logs) into its cache with the key `78awdk123` (the computed hash)

Reply:
I prefer my version.